scripts/ci/discover_aws_runner_config.py

```
@@ -0,0 +1,242 @@
+
+    Args:
+        region: AWS region name.
+        args: Arguments placed after ``aws ec2``.
+
+    Returns:
+        Parsed JSON output, or ``None`` when the command fails or returns no
+        output.
+    """
+    cmd = ["aws", "ec2", *args, "--region", region, "--output", "json"]
+    try:
+        completed = subprocess.run(cmd, check=True, capture_output=True, text=True)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)